File: frontend/react-paylateapp/src/models/AddProductRequest.ts
class AddProductRequest {
    productName: string;
    price: number;
    description: string;
    category: string;
    quantity: number;
    imgUrl?: string;
    rating: number;
    authorName: string;
    authorUrl: string;
    productType: string;

    constructor(productName: string, price: number, description: string,
        category: string, quantity: number, imgUrl: string, rating: number, 
        authorName: string, authorUrl: string, productType: string){
        this.productName = productName;
        this.price = price;
        this.description = description;
        this.category = category;
        this.quantity = quantity;
        this.imgUrl = imgUrl;
        this.rating = rating;
        this.authorName = authorName;
        this.authorUrl = authorUrl;
        this.productType = productType;
    }
}

export default AddProductRequest;

File: backend/admin-service/src/main/java/org/payLate/requestModels/AddProductRequest.java
package org.payLate.requestModels;

import lombok.Data;

@Data
public class AddProductRequest {

    private String productName;

    private Double price;

    private String description;

    private String category;

    private Long quantity;

    private String imgUrl;

    private Double rating;

    private String authorName;

    private String authorUrl;

    private String productType;
}
